src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CollectionSettings.java

```diff
@@ -179,49 +258,112 @@ private static CollectionSettings createCollectionSettings(
       SimilarityFunction function,
       String comment,
       ObjectMapper objectMapper) {
-
     if (comment == null || comment.isBlank()) {
-      return new CollectionSettings(
-          collectionName, vectorEnabled, vectorSize, function, null, null, null);
+      if (vectorEnabled) {
+        return new CollectionSettings(
+            collectionName, new VectorConfig(true, vectorSize, function, null), null);
+      } else {
+        return new CollectionSettings(collectionName, VectorConfig.notEnabledVectorConfig(), null);
+      }
     } else {
-      String vectorizeServiceName = null;
-      String modelName = null;
-      JsonNode commentConfig;
+      JsonNode commentConfigNode;
       try {
-        commentConfig = objectMapper.readTree(comment);
+        commentConfigNode = objectMapper.readTree(comment);
       } catch (JsonProcessingException e) {
         // This should never happen, already check if vectorize is a valid JSON
         throw new RuntimeException("Invalid json string, please check 'options' configuration.", e);
       }
-      JsonNode vectorizeConfig = commentConfig.path("vectorize");
-      if (!vectorizeConfig.isMissingNode()) {
-        vectorizeServiceName = vectorizeConfig.path("service").textValue();
-        JsonNode optionsNode = vectorizeConfig.path("options");
-        modelName = optionsNode.path("modelName").textValue();
-        if (!(vectorizeServiceName != null
-            && !vectorizeServiceName.isEmpty()
-            && modelName != null
-            && !modelName.isEmpty())) {
-          // This should never happen, VectorizeConfig check null, unless it fails
-          throw new JsonApiException(
-              VECTORIZECONFIG_CHECK_FAIL,
-              "%s, please check 'vectorize' configuration."
-                  .formatted(VECTORIZECONFIG_CHECK_FAIL.getMessage()));
+
+      // new table comment design, with collection as top-level key
```

Comment:
Can you create an interface and have different implementation for schema version 1 and schema version missing? This way when we come up with next version changes will be minimum.